Marlin/ultralcd.cpp

```diff
@@ -450,6 +450,12 @@ void lcd_set_home_offsets() {
   static void _lcd_babystep(int axis, const char *msg) {
     if (encoderPosition != 0) {
       babystepsTodo[axis] += (int)encoderPosition;
+      baby_max_endstop[axis] -= ((int)encoderPosition)/axis_steps_per_unit[axis];
+      baby_min_endstop[axis] -= ((int)encoderPosition)/axis_steps_per_unit[axis];
+     #ifdef BABYSTEP_OFFSET
+      if(axis == Z_AXIS)
+        home_offset[axis] += ((int)encoderPosition)/axis_steps_per_unit[axis];
+     #endif //BABYSTEP_OFFSET
```

Comment:
Call me old-fashioned, but I like doing repeated calculations up-front even though the compiler will probably optimize it into the very same code. In this case it's a little more readable too.
```
      float diff = int(encoderPosition) / axis_steps_per_unit[axis];
      baby_max_endstop[axis] -= diff;
      baby_min_endstop[axis] -= diff;
      #ifdef BABYSTEP_OFFSET
        if (axis == Z_AXIS) home_offset[Z_AXIS] += diff;
      #endif //BABYSTEP_OFFSET
```